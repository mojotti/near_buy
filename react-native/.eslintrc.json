{
  "extends": "airbnb",
  "rules": {
    "func-names": ["error", "never"],
    "indent": ["error", 2],
    "arrow-parens": "off",
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
    "react/jsx-indent": ["error", 2],
    "react/jsx-indent-props": ["error", 2],
    "import/no-named-as-default": 0,
    "import/prefer-default-export": 0,
    "no-undef": "off",
    "no-use-before-define": "off"
  },
  "env": {
    "jest": true
  },
  "globals": { "fetch": false }
}
