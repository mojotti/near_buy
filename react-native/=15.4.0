NearBuy@0.1.0 /Users/ristomatti/PycharmProjects/near_buy/react-native
└── UNMET PEER DEPENDENCY react@16.0.0

