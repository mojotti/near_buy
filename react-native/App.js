import React from 'react';
import { Provider } from 'react-redux';
import Application from './components/Application';
import store from './redux';

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Application />
      </Provider>
    );
  }
}
