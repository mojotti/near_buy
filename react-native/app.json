{
  "name": "NearBuy",
  "displayName": "NearBuy",
  "expo": {
    "sdkVersion": "25.0.0"
  }
}