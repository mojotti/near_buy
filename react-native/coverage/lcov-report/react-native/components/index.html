<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-native/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> react-native/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/89</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.79% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>14/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>37/89</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Items.js"><a href="Items.js.html">Items.js</a></td>
	<td data-value="41.18" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="20" class="abs low">6/20</td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="41.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="20" class="abs medium">15/20</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 18 2017 21:32:17 GMT+0300 (EEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
