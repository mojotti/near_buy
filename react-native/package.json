{
  "name": "NearBuy",
  "version": "0.1.0",
  "private": true,
  "devDependencies": {
    "@types/js-base64": "^2.1.5",
    "babel-core": "^6.26.3",
    "babel-eslint": "8.2.6",
    "babel-preset-es2015-mod": "^6.6.0",
    "babel-preset-es3": "^1.0.1",
    "babel-preset-react-native-stage-0": "^1.0.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "enzyme-to-json": "^3.3.4",
    "eslint": "4.19.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-jsx-a11y": "^6.0.2",
    "eslint-plugin-react": "^7.4.0",
    "flow-bin": "^0.49.1",
    "jest": "^23.3",
    "jest-cli": "^21.2.1",
    "prettier": "1.11.0",
    "react-native-mock": "^0.3.1",
    "react-test-renderer": "16.2.0",
    "redux-devtools": "^3.4.0",
    "redux-mock-store": "^1.3.0",
    "regenerator-runtime": "^0.12.1",
    "sinon": "^4.0.1"
  },
  "scripts": {
    "start": "react-native start",
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "test": "jest",
    "test:watch": "npm test -- --watch",
    "flow": "flow",
    "clean-install": "rm -rf node_modules && rm -rf $TMPDIR/react-* && yarn cache clean --force && watchman watch-del-all && yarn"
  },
  "jest": {
    "testEnvironment": "node",
    "preset": "react-native",
    "setupTestFrameworkScriptFile": "<rootDir>/test-setup.js",
    "transformIgnorePatterns": [
      "node_modules/(?!react-native|react-navigation)/"
    ],
    "setupFiles": [
      "<rootDir>/.config/jest/globalFetch.js"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "testPathIgnorePatterns": [
      "/node_modules/",
      "<rootDir>/__tests__/__mocks__/"
    ]
  },
  "dependencies": {
    "babel-jest": "^23.4.2",
    "babel-preset-expo": "^4.0.0",
    "babel-preset-react-native": "5.0.0",
    "base-64": "^0.1.0",
    "isomorphic-fetch": "^2.2.1",
    "lottie-react-native": "^2.5.6",
    "prop-types": "^15.6.1",
    "react": "16.4.1",
    "react-dom": "^16.0.0",
    "react-native": "0.56.0",
    "react-native-fetch-blob": "^0.10.8",
    "react-native-image-crop-picker": "^0.20.3",
    "react-native-image-placeholder": "^1.0.14",
    "react-native-looped-carousel": "^0.1.10",
    "react-native-maps": "^0.21.0",
    "react-native-modal": "^6.5.0",
    "react-native-snap-carousel": "^3.7.1",
    "react-native-swipeout": "^2.3.3",
    "react-native-vector-icons": "^4.5.0",
    "react-navigation": "^1.0.0-beta.13",
    "react-redux": "^5.0.6",
    "redux": "^3.7.2",
    "redux-persist": "^5.9.1",
    "redux-persist-transform-encrypt": "^2.0.0",
    "redux-thunk": "^2.3.0"
  }
}
