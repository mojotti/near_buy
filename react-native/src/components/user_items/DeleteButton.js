import React, { Component } from 'react';
import { Alert, Text, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';

import { styles } from '../../static/styles/UserItemDetailsStyles';
import { DELETION_ERROR, localhost } from '../../static/constants';
import { getBearerHeaders } from '../../networking/networking';

export default class DeleteButton extends Component {
  _navigateBack = () => {
    const { goBack } = this.props.navigation;
    goBack();
  };

  _removeItemAndNavigateBack = () => {
    const url = `http://${localhost}:5000/api/v1.0/user/items/${this.props.id}`;
    fetch(url, {
      method: 'DELETE',
      headers: getBearerHeaders(this.props.token),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.ok) {
          this.props.fetchItems();
          this._navigateBack();
        }
      })
      .catch(error => {
        Alert.alert(...DELETION_ERROR);
        console.error(error);
      });
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this._removeItemAndNavigateBack}
        style={styles.deleteButton}
      >
        <Text style={styles.buttonText}>Delete item</Text>
      </TouchableHighlight>
    );
  }
}

DeleteButton.propTypes = {
  id: PropTypes.number.isRequired,
  token: PropTypes.string.isRequired,
};
