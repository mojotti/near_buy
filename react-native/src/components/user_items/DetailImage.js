import React, { Component } from 'react';
import { Alert, Dimensions, TouchableHighlight, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import ImagePicker from 'react-native-image-crop-picker';
import PropTypes from 'prop-types';
import RNFetchBlob from 'react-native-fetch-blob';
import { getFormDataHeaders } from '../../networking/networking';
import { localhost } from '../../static/constants';
import { connect } from 'react-redux';
import ImagePlaceholder from '../image_placeholder/ImagePlaceholder';
const { height, width } = Dimensions.get('window');

export class _DetailImage extends Component {
  constructor(props) {
    super(props);
    this.handleImageSelection = this.handleImageSelection.bind(this);
    this.handleNewImage = this.handleNewImage.bind(this);
  }

  renderImage = () => {
    return (
      <View style={{ flex: 1 }}>
        <ImagePlaceholder
          url={this.props.url}
          styles={{ width, height: width, paddingTop: 20 }}
          placeholderSize={width * 0.8}
        />
      </View>
    );
  };

  sendItemToDb = image => {
    let images = [];
    images.push({
      name: 'pictures[]',
      filename: 'image' + this.props.numOfPics + '.jpg',
      type: image.mime,
      data: RNFetchBlob.wrap(image.path),
    });

    const url = `http://${localhost}:5000/api/v1.0/items/${
      this.props.id
    }/add_picture`;
    RNFetchBlob.fetch('POST', url, getFormDataHeaders(this.props.token), images)
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.ok) {
          this.props.onImageUpload();
        }
      })
      .catch(error => {
        console.error(error);
      });
  };

  handleNewImage = () => {
    ImagePicker.openCamera({
      width: 400,
      height: 400,
      cropping: true,
    })
      .then(image => this.sendItemToDb(image))
      .catch(error => console.log('error in image selection:', error));
  };

  handleGalleryImage = () => {
    ImagePicker.openPicker({
      width: 400,
      height: 400,
      cropping: true,
    })
      .then(image => this.sendItemToDb(image))
      .catch(error => console.log('error in image selection:', error));
  };

  handleImageSelection() {
    Alert.alert(
      'Select picture',
      'Select from gallery or take a new one with camera',
      [
        {
          text: 'Cancel',
          onPress: () => {},
          style: 'cancel',
        },
        {
          text: 'Gallery',
          onPress: () => this.handleGalleryImage(),
        },
        { text: 'Camera', onPress: () => this.handleNewImage() },
      ],
      { cancelable: false }
    );
  }

  renderPlaceholder = () => {
    return (
      <View
        style={{
          flex: 1,
          alignSelf: 'center',
          alignItems: 'center',
          justifyContent: 'center',
          paddingBottom: 40,
          backgroundColor: '#FFFFFF',
          height: width,
          width: width,
        }}
      >
        <TouchableHighlight
          onPress={this.handleImageSelection}
          underlayColor={'#888888'}
        >
          <MaterialIcons name="add-a-photo" size={height * 0.4} color="gray" />
        </TouchableHighlight>
      </View>
    );
  };

  render() {
    return this.props.url ? this.renderImage() : this.renderPlaceholder();
  }
}

_DetailImage.propTypes = {
  url: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  token: PropTypes.string.isRequired,
  onImageUpload: PropTypes.func.isRequired,
  numOfPics: PropTypes.number.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const token = state.authorizationReducer.token;
  return { token };
};

const DetailImage = connect(mapStateToProps, null)(_DetailImage);
export default DetailImage;
