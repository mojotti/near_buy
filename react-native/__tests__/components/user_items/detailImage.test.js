import React from 'react';
import { Alert } from 'react-native';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { _DetailImage } from '../../../src/components/user_items/DetailImage';

jest.mock('react-native-image-crop-picker', () => {
  return {
    openPicker: jest.fn(() => Promise.resolve('./image.jpg')),
    openCamera: jest.fn(() => Promise.resolve('./image.jpg')),
  };
});

describe('<DetailImage />', () => {
  beforeEach(() => {
    Alert.alert = jest.fn();
  });

  test('renders correctly', () => {
    const detailImage = shallow(
      <_DetailImage
        url="foo.bar.png"
        id={1}
        token="foo"
        onImageUpload={() => {}}
        numOfPics={1}
      />
    );

    expect(detailImage).toMatchSnapshot();
  });

  test('renders placeholder when url is not given', () => {
    const detailImage = shallow(
      <_DetailImage
        url=""
        id={1}
        token="foo"
        onImageUpload={() => {}}
        numOfPics={1}
      />
    );

    expect(detailImage).toMatchSnapshot();
  });

  test('placeholder handles image selection on press', () => {
    const detailImage = shallow(
      <_DetailImage
        url=""
        id={1}
        token="foo"
        onImageUpload={() => {}}
        numOfPics={1}
      />
    );

    detailImage
      .dive()
      .find('TouchableHighlight')
      .simulate('Press');

    expect(Alert.alert.mock.calls[0][0]).toEqual('Select picture');
    expect(Alert.alert.mock.calls[0][1]).toEqual(
      'Select from gallery or take a new one with camera'
    );
    expect(Alert.alert.mock.calls[0][2]).toMatchSnapshot();
  });

  test('handleNewImage sends item to db on success', async () => {
    const detailImage = shallow(
      <_DetailImage
        url=""
        id={1}
        token="foo"
        onImageUpload={() => {}}
        numOfPics={1}
      />
    );

    const sendItemToDbSpy = jest.spyOn(detailImage.instance(), 'sendItemToDb');
    await detailImage.instance().handleNewImage();

    expect(sendItemToDbSpy.mock.calls.length).toEqual(1);
    expect(sendItemToDbSpy.mock.calls[0][0]).toEqual('./image.jpg');
  });
});
