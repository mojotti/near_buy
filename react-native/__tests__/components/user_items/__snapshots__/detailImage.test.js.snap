// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailImage /> placeholder handles image selection on press 1`] = `
Array [
  Object {
    "onPress": [Function],
    "style": "cancel",
    "text": "Cancel",
  },
  Object {
    "onPress": [Function],
    "text": "Gallery",
  },
  Object {
    "onPress": [Function],
    "text": "Camera",
  },
]
`;

exports[`<DetailImage /> renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ImagePlaceholder
    placeholderSize={600}
    styles={
      Object {
        "height": 750,
        "paddingTop": 20,
        "width": 750,
      }
    }
    url="foo.bar.png"
  />
</Component>
`;

exports[`<DetailImage /> renders placeholder when url is not given 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "height": 750,
      "justifyContent": "center",
      "paddingBottom": 40,
      "width": 750,
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    underlayColor="#888888"
  >
    <Icon
      allowFontScaling={false}
      color="gray"
      name="add-a-photo"
      size={533.6}
    />
  </TouchableHighlight>
</Component>
`;
